src/shared/components/query/GeneSetSelector.tsx

```diff
@@ -60,18 +62,14 @@ export default class GeneSetSelector extends QueryStoreComponent<{}, {}>
 	    ];
 	}
 
-	@computed get textAreaRef()
-	{
-		if (this.store.geneQueryErrorDisplayStatus === 'shouldFocus')
-			return (textArea:HTMLTextAreaElement) => {
-				let {error} = this.store.oql;
-				if (textArea && error)
-				{
-					textArea.focus();
-					textArea.setSelectionRange(error.start, error.end);
-					this.store.geneQueryErrorDisplayStatus = 'focused';
-				}
-			};
+	@action
```

Comment:
this method also needs to be @bind .  this could explain failure to update store.   yes, i see it doesn't have context (this).